map.js

```diff
@@ -467,66 +415,108 @@ function initListeners(){
             moveMap({x:event.movementX, y:event.movementY});
         }
         // regardless of whether we are down or not, we need to redraw the scene?
-        // TODO: only redraw if we move on to or off of an icon.
+        // TODO: only redraw if we dragged or move on to or off of an icon?
         drawFrame();
```

Comment:
I moved drawFrame() out of most of the updateXXX() methods so its easier to ensure that we only call it once per update.